python/thunder/rdds/images.py

```diff
@@ -436,6 +436,41 @@ def medianFilter(self, size=2):
 
         return self.filter(func=lambda x: median_filter(x, size))
 
+    def localCorr(self, neighborhood=2):
+        """
+        Correlate every pixel to the average of its local neighborhood.
+
+        This algorithm computes, for every spatial record, the correlation coefficient
+        between that record's series, and the average series of all records within
+        a local neighborhood with a size defined by the neighborhood parameter.
+        For data with three spatial keys, only neighborhoods in x and y
+        currently supported.
+
+        parameters
+        ----------
+        neighborhood: int, optional, default=2
+            Size of the correlation neighborhood in pixels. A sequence is interpreted
+            as the neighborhood size for each axis. For three-dimensional data, a single
+            scalar is intrepreted as the neighborhood in x and y, with no filtering in z.
+        """
+        from numpy import corrcoef
+
+        nimages = self.nrecords
+
+        # Spatially average the original image set over the specified neighborhood
+        blurred = self.uniformFilter((neighborhood * 2) + 1)
```

Comment:
Add another step after this that uses `applyKeys` to increment the keys by nimages, that way the keys will already be correct in the next step and we don't need to use `renumber()`.